WordPress/openverse
# Collect image dimensions from Europeana

## Fixes
Fixes #1484 by @stacimc 

## Description

Adds image dimensions, filetype, and filesize from Europeana item data.

## Testing Instructions

At a high level, the goal is to run all the tests using `pytest`, then run the dag in airflow, then look at the data to confirm that it looks reasonable. 

- Run `just catalog/test-session` and then `pytest tests/dags/providers/provider_api_scripts/test_europeana.py`. 
- When I ran `pytest` alone from within the same test session, I got a bunch of errors from `test_ingestion_server.test_index_readiness_check` but that's just because the ingestion server stuff isn't started up when you're just running docker for a catalog test session.
- To get the local port for airflow (http://localhost:9090/home) working in docker, I ran `just catalog/recreate`. Then you can turn on the dag and it will go through daily runs and be sure to pick up some data. Check out the results using sql after `just catalog/pgcli`.

I temporarily replaced the dag start date with 8/8/2023 in provider_workflows.py, to get a smaller sample size for testing.  Only two days out of nine had data, but it seems like the 3 second timeout might be reasonable at least for the daily runs, if these numbers of records are comparable with production. I would need to know the total number of records in production to figure out if that is workable for a refresh.

```
*DAG*: `europeana_workflow`
*Date range*: 2023-08-08 -> 2023-08-09
*Duration of data pull tasks*: 24 mins
*Number of records upserted per media type*:
  - `image`: 474

*DAG*: `europeana_workflow`
*Date range*: 2023-08-15 -> 2023-08-16
*Duration of data pull tasks*: 56 mins, 20 secs
*Number of records upserted per media type*:
  - `image`: 1,113
```

The new fields are populated in all of the inserted records:

```
openledger> select filetype, count(*) records, count(filesize) filesize, count(width) width, count(height) height from image group by filetype order by 
 2 desc;
+----------+---------+----------+-------+--------+
| filetype | records | filesize | width | height |
|----------+---------+----------+-------+--------|
| jpg      | 1573    | 1573     | 1573  | 1573   |
| png      | 14      | 14       | 14    | 14     |
+----------+---------+----------+-------+--------+
SELECT 2
Time: 0.020s
```


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
